src/test/s3/commands/uploadFile.test.ts

```diff
@@ -58,31 +197,182 @@ describe('uploadFileCommand', function () {
+        await uploadFileCommand(instance(s3), fileLocation, statFile, getBucket, getFile, window, outputChannel, commands)
+         
+        assert.ok(window.message.error?.includes('Failed to upload file'))
     })
+    
+    
+})
 
-    it('does nothing when prompt is cancelled', async function () {
-        await uploadFileCommand(node, statFile, new FakeWindow(), new FakeCommands())
+describe('getFileToUpload', function () {
+    const fileLocation = vscode.Uri.file('/file.jpg')    
+    let window: FakeWindow
 
-        verify(s3.uploadFile(anything())).never()
+    const selection: any = {label: "Browse for more files..." }
+    const prompt: <T extends vscode.QuickPickItem>(opts: {
+        picker: vscode.QuickPick<T>
+        onDidTriggerButton?(
+            button: vscode.QuickInputButton,
+            resolve: (value: T[] | PromiseLike<T[] | undefined> | undefined) => void,
+            reject: (reason?: any) => void
+        ): void
+    }) => Promise<T[] | undefined> = () => {
+        return new Promise((resolve, reject) => {
+            resolve([selection])
+        })
+    }
+
+    beforeEach(function () {
+        window = new FakeWindow({ dialog: { openSelections: [fileLocation] } })       
+    })
+    
+    it("directly asks user for file if no active editor", async function () {
+        
+        const response = await getFileToUpload(undefined, window, prompt)
+        assert.strictEqual(response, fileLocation)
     })
 
-    it('shows an error message when upload fails', async function () {
-        when(s3.uploadFile(anything())).thenReject(new Error('Expected failure'))
+    it("Returns undefined if no file is selected on first prompt", async function () {
+        window = new FakeWindow({ dialog: { openSelections: undefined } })
+        
+        const response = await getFileToUpload(undefined, window, prompt)
+        assert.strictEqual(response, undefined)
+    })
 
-        const window = new FakeWindow({ dialog: { openSelections: [fileLocation] } })
-        const commands = new FakeCommands()
-        await uploadFileCommand(node, statFile, window, commands)
+    it("prompts quick pick with open file option, and browse for more files, returns open file if that option is selected", async function(){
+        selection.label = fileLocation.fsPath
+        const response = await getFileToUpload(fileLocation, window, prompt)
 
-        assert.ok(window.message.error?.includes('Failed to upload file'))
+        assert.strictEqual(response, fileLocation)
+    })
+
+    it("prompts quick pick with open file option, and browse for more files, prompts user for other file if that option is selected", async function (){
+        selection.label = "Browse for more files..."
+        
+        const response = await getFileToUpload(fileLocation, window, prompt)
+        assert.strictEqual(response, fileLocation)
+
+    })
+
+    it("returns undefined on second prompt if no file is selected", async function () {
```

Comment:
`'returns undefined if the user does not select a file through the file browser'` ("second prompt" is vague and can potentially change)